discovery/utils.go

```diff
@@ -0,0 +1,92 @@
+package discovery
+
+import (
+	"encoding/binary"
+
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/lnwire"
+)
+
+// waitingProofKey is the proof key which uniquely identifies the
```

Comment:
This definition should be moved to the top of `service.go`. As it stands now, the placement of this definition increases the amount of work for a reader as it forces them to jump jump around to another file just to obtain the definition of a type that's only used in a single file.